scipy/sparse/tests/test_base.py

```diff
@@ -4137,6 +4137,25 @@ def test_binop_explicit_zeros(self):
         assert (a + b).nnz == 2
         assert a.multiply(b).nnz == 1
 
+    def test_broadcast_to(self):
+        a = np.array([[1, 0, 2]])
+        b = np.array([[1], [0], [2]])
+        c = np.array([[1, 0, 2], [0, 3, 0]])
+        d = np.array([[7]])
+        e = np.array([[0]])
+        f = np.array([[0,0,0,0]])
+        res_a = csr_matrix(a).broadcast_to((2,3))
```

Comment:
Some of these tests should check that a ValueError occurs when the shape doesn't match.  In my playing with it,  it looks like using `np.broadcast_shapes` isn't what we want. It checks if the shapes can be broadcast to a unified shape. We need some tests to catch that.  Perhaps something like:
```python
    with pytest.raises(ValueError, match="cannot be broadcast"):
        csr_matrix([[1, 2, 3], [3, 2, 1]], (2, 1))
```
Note that `np.broadcast_shape((2, 3), (2, 1))` returns `(2, 3)` without raising. So I think your original check is needed.

Reply:
Ah right, got it. I've reverted the change and added tests to ensure that a ValueError is raised (96a3c54910b563bba252ca0c728af2295c8ad7ff).